src/regdesk/states/new_search_state.js

```diff
@@ -0,0 +1,65 @@
+import { RegState } from './reg_state.js';
+
+/**
+ * New Search state
+ */
+export class NewSearchState extends RegState {
+  /**
+   * Get the transition events this state can process.
+   */
+  static get events() {
+    // TODO: This is apparently how TS does enums, make it an enum someday?
+    return {
+      START_SEARCH: 'START_SEARCH',
+      RESET: 'RESET',
+    };
+  }
+
+  /**
+   * Initializes a new instance of the NewSearch class.
+   * @param {RegMachineArgs} regMachineArgs - Standard argument set for states.
+   * @param {*} eventMap - Mapping of this object's transition events to new states.
+   */
+  constructor(regMachineArgs, eventMap) {
+    super(regMachineArgs, eventMap);
+  }
+
+  /**
+   * Enter this state.
+   */
+  enterState() {
+    this.leftButton.hide();
+    this.rightButton.hide();
+
+    this.centerField.innerHTML = this.#getCenterTemplate();
+    this.centerField.querySelector('#newSearchText').focus();
+
+    // Search isn't wired up right now so instead we'll just transition
+    // to the loading screen.
+    this.centerField.querySelector('#newSearchButton')
+      .addEventListener('click', (e) => {
+        // Just to demonstrate the setup, this would talk to the commMgr
+        // and actually run the search.
+        // TODO: Should the event objects be strongly typed? Correlated somehow?
+        const searchText = this.centerField.querySelector('#newSearchText').value;
+        this.dispatchTransition(NewSearchState.events.START_SEARCH, { searchText: searchText });
+      });
+  }
+
+  /**
+   *
+   * @return {string} - The center html elements.
+   */
+  #getCenterTemplate() {
```

Comment:
Kind of like translations, your javascript shouldn't really be injecting html either. Just hiding/showing the correct fields. That way all of your HTML is in one place, nicely separated from your javascript. And if you need to change one, you don't have to worry if you missed something in the other.

Reply:
Ahha this brings the previous comment into better focus. Let me see if I can migrate stuff over.

Reply:
See above, no more injection.